synapse/handlers/room_member.py

```diff
@@ -210,12 +211,19 @@ def _update_membership(
             if is_blocked:
                 raise SynapseError(403, "This room has been blocked on this server")
 
-        if (effective_membership_state == "invite" and
-                self.hs.config.block_non_admin_invites):
-            is_requester_admin = yield self.auth.is_server_admin(
-                requester.user,
-            )
-            if not is_requester_admin:
+        if effective_membership_state == "invite":
+            block_invite = False
+            if self.hs.config.block_non_admin_invites:
+                is_requester_admin = yield self.auth.is_server_admin(
+                    requester.user,
+                )
+                if not is_requester_admin:
+                    block_invite = True
+
+            if not self.spam_checker.user_may_invite(requester.user):
+                block_invite = True
```

Comment:
Do we want this to apply to server admins?

Reply:
hmm, I guess not actually